tsup.config.ts

```diff
@@ -0,0 +1,13 @@
+import { defineConfig } from 'tsup'
+
+export default defineConfig({
+  entry: ['src/index.ts'],
+  clean: true,
+  dts: true,
+  format: ['esm', 'cjs'],
+  minify: true,
+  sourcemap: true,
+  splitting: true,
+  target: 'node16',
```

Comment:
I think Node.js 16 is the lower bound in the current "works but deprecated" sense. You should be able to easily update the lower bound to Node.js 20 by simply changing this in the future.